ouroboros-network/src/Ouroboros/Network/PeerSelection/LedgerPeers.hs

```diff
@@ -290,3 +291,34 @@ runLedgerPeers inRng tracer readUseLedgerAfter LedgerPeersConsensusInterface{..}
     splitPeers (addrs, domains) (RelayAddress ip port) =
         let !addr = IP.toSockAddr (ip, port) in
         (Set.insert addr addrs, domains)
+
+
+-- | For a LederPeers worker thread and submit request and receive responses.
+--
+withLedgerPeers :: forall m a.
```

Comment:
Let's use `withLedgerPeers` in the tests, and do not expose other low level functions from this module.

Reply:
done

Reply:
done